builder/azure/arm/config.go

```diff
@@ -362,6 +368,23 @@ func setCustomData(c *Config) error {
 	return nil
 }
 
+func setSasUrlDuration(c *Config) error {
+	if c.SasUrlLifetime == "" {
+		c.SasUrlLifetime = "720h"
+	}
+	match := regexp.MustCompile("^([0-9]+[hm])([0-9]+[m])?$")
+	if match.MatchString(c.SasUrlLifetime) != true {
+		err := fmt.Errorf("Failed to convert SasUtlLifetime to duration, format is invalid: %s", c.SasUrlLifetime)
+		return err
+	}
+	b, err := time.ParseDuration(c.SasUrlLifetime)
```

Comment:
Why do you need to match a regular expression, and use ParseDuration.  ParseDuration is sufficient, and will be more robust.